src/types/redis_cms.cc

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "redis_cms.h"
+
+#include <stdint.h>
+
+#include "cms.h"
+
+namespace redis {
+
+rocksdb::Status CMS::GetMetadata(engine::Context &ctx, const Slice &ns_key,
+                                 CountMinSketchMetadata *metadata) {
+  return Database::GetMetadata(ctx, {kRedisCountMinSketch}, ns_key, metadata);
+}
+
+rocksdb::Status CMS::IncrBy(engine::Context &ctx, const Slice &user_key, const std::unordered_map<std::string, uint64_t> &elements) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+  CountMinSketchMetadata metadata{};
```

Comment:
If not exists, does this mean getting a `<0, 0, 0>` pair? What does it means?

Reply:
Yeah, you would get a 0,0,0 pair